doc/en/components/grids/_shared/conditional-cell-styling.md

```diff
@@ -0,0 +1,684 @@
+
+
+Let's define our styles:
+
+```typescript
+// component.ts
+public oddColStyles = {
+    background: 'linear-gradient(to right, #b993d6, #8ca6db)',
+    color: (rowData, coljey, cellValue, rowIndex) => rowIndex % 2 === 0 ? 'white' : 'gray',
+    animation: '0.75s popin'
+};
+
+public evenColStyles = {
+    background: 'linear-gradient(to right, #8ca6db, #b993d6)',
+    color: (rowData, coljey, cellValue, rowIndex) => rowIndex % 2 === 0 ? 'gray' : 'white',
+    animation: '0.75s popin'
+};
+```
+
+```razor
+define styles for blazor
+```
+
+<!-- Angular -->
+On `ngOnInit` we will add the `cellStyles` configuration for each column of the predefined `columns` collection, which is used to create the {ComponentName} columns dynamically.
+
+```ts
+// component.ts
+public ngOnInit() {
+    this.data = athletesData;
+    this.columns = [
+        { field: 'Id' },
+        { field: 'Position' },
+        { field: 'Name' },
+        { field: 'AthleteNumber' },
+        { field: 'CountryName' }
+    ];
+
+    this.applyCSS();
+}
+
+public applyCSS() {
+    this.columns.forEach((column, index) => {
+        column.cellStyles = (index % 2 === 0 ? this.evenColStyles : this.oddColStyles);
+    });
+}
+
+public updateCSS(css: string) {
+    this.oddColStyles = {...this.oddColStyles, ...JSON.parse(css)};
+    this.evenColStyles = {...this.evenColStyles, ...JSON.parse(css)};
+    this.applyCSS();
+}
+```
+
+```html
+// component.html
+<igx-grid
+    #grid1 [data]="data"
+    primaryKey="ID"
+    width="80%"
+    height="300px">
+    <igx-column *ngFor="let c of columns"
+        [field]="c.field"
+        [header]="c.field"
+        [cellStyles]="c.cellStyles">
+    </igx-column>
+</igx-grid>
+```
+
+Define a `popin` animation
+
+```scss
+// component.scss
+@keyframes popin {
+    0% {
+        opacity: 0.1;
+        transform: scale(.75, .75);
+        filter: blur(3px) invert(1);
+    }
+
+    50% {
+        opacity: .5;
+        filter: blur(1px);
+    }
+
+    100% {
+        transform: scale(1, 1);
+        opacity: 1;
+        filter: none;
+    }
+}
+```
+<!-- end: Angular -->
+
+<!-- Blazor -->
+Add similar to ngOnInit
+Add similar handling
+<!-- end: Blazor -->
+
+### Demo
+
+<!-- ComponentStart: Grid -->
+
+<code-view style="height:620px"
+           data-demos-base-url="{environment:demosBaseUrl}"
+           iframe-src="{environment:demosBaseUrl}/{ComponentSample}-cell-cellStyling"
+           github-src="{ComponentSample}/cell-cellStyling" >
+</code-view>
+
+<!-- ComponentEnd: Grid -->
+
+
+<!-- ComponentStart: TreeGrid -->
+
+<code-view style="height:600px"
+           data-demos-base-url="{environment:demosBaseUrl}"
+           iframe-src="{environment:demosBaseUrl}/{ComponentSample}-cell-cellStyling"
+           github-src="{ComponentSample}/cell-cellStyling" >
+</code-view>
+
+<!-- ComponentEnd: TreeGrid -->
+
+<!-- ComponentStart: HierarchicalGrid -->
+
+<!-- TODO -->
+
+<!-- ComponentEnd: HierarchicalGrid -->
```

Comment:
replace above 3 code-view blocks with just one code-view and remove component build flag because routing path to the sample will be the same for all grid samples in Sample Browser.